### DISCUSSION TITLE
menu_add

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#1252] [Ideas / Feature Requests]
Can we implement this by adding navigation items directly to the Menu, like menu_add_item('label').props.classes()?

I think features like this  are generic and basic.It's a little complicate to implenment it in the way like in the websit

![微信截图_20230725090252](https://github.com/zauberzeug/nicegui/assets/25096045/70419705-6e0c-4b0a-8208-83e91c4066ef)

**Comment by rodja** [comment]
I'm sorry @v479038280, I do not understand. Could you elaborate on `menu_add_item('label').props.classes()`? Maybe show some code which would create the look from the screenshot. How is it done now? And how would you suggest to improve the API?

**Reply by v479038280** [reply]
menu_add_item('label').props.classes()   is just a text ?  we couldn't use it to navigate between the pages and its content?

**Reply by rodja** [reply]
I still have no clue about what you want to archive. In fact, it is even more unclear now. You write "it's just a text". Do you mean a string that should be parsed? What is the goal? Can you please be more specific and give a concrete example?

**Reply by v479038280** [reply]
When we use menu_add_item('HOME').props.classes(), and click the label  'HOME' in the UI ,can we redirect to Home page ,or it's just a label,couldn't attatch other attribute.

**Reply by v479038280** [reply]
The website implement it this way 

![微信截图_20230725111706](https://github.com/zauberzeug/nicegui/assets/25096045/d1a27a68-4078-4674-9930-df944e67e352)

**Reply by rodja** [reply]
`menu_add_item('HOME').props.classes()` is not valid syntax. Do you mean `ui.menu`? In combination with `ui.open` you can surely open another webpage:

```py
with ui.button(icon='menu'):
    with ui.menu() as menu:
        ui.menu_item('Homepage', lambda: ui.open('https://nicegui.io/'))
        ui.menu_item('Docs', lambda: ui.open('https://nicegui.io/documentation'))
        ui.menu_item('GitHub', lambda: ui.open('https://github.com/zauberzeug/nicegui'))
```

But what has that to do with how the website is implemented? `load_demo` is a helper function...

**Reply by v479038280** [reply]
Thank you I'll try ui.open().

**Comment by falkoschindler** [comment]
I'll close this feature request, because it's purpose remains unclear.
